SQL_module/SubjectForm_Task2/db_manipulation.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+    include "./db_connection.php";
+
+    $table_name = "electives";
+
+    function insertInTable($title, $description, $lecturer){
+        try {
+
+            $connection = db_connection();
+            $sql = "INSERT INTO electives(title,description,lecturer) VALUES (:title, :description, :lecturer)";
+    
+            $preparedSql = $connection->prepare($sql) or die("Failed to prepare sql query.");
+            $preparedSql->bindParam(':title', $title);
+            $preparedSql->bindParam(':description', $description);
+            $preparedSql->bindParam(':lecturer', $lecturer);
+            
+            $preparedSql->execute() or die("Filed to execute sql query.");   
```

Comment:
You may want to proactively close the sql connections when you finish with them. 
https://www.w3schools.com/php/php_mysql_connect.asp

https://stackoverflow.com/a/15444763/3837638
> it's not pointless to close a connection. Every connection occupies resources in the database server. Most servers are configured with a maximum number of concurrent connections. If you close connections proactively, you free up that slot for another thread, and you get overall better throughput.

Reply:
This applies to all places where you create a new connection and use it.